File: json_to_flutter/models/submodels/shadow.dart
import '../../functions/util.dart';
import '../../main.dart';

class Shadow {
  double x;
  double y;
  String color;
  double opacity;
  bool visible;
  double blurRadius;

  Shadow(
      {this.x,
      this.y,
      this.color,
      this.opacity,
      this.visible,
      this.blurRadius});

  Shadow.fromJson(Map<String, dynamic> json, bool withColor) {
    visible = json['visible'] && withColor;
    if (visible) {
      x = fixDouble(json['x']);
      y = fixDouble(json['y']);
      opacity = fixDouble(json['opacity']);
      color = hexColorToFlutterColor(json['color'], opacity);
      blurRadius = fixDouble(json['blur']);
    }
  }
  String print({bool isText = false}) {
    if (visible) {
      String offSet = "offset: Offset(${sz(x)},${sz(y)})";
      String blurR = "${sz(blurRadius)}";
      if (withDivision)
        return """
        ..boxShadow(${color}, blur: $blurR, $offSet)
        """;
      return """
      ${isText ? "shadows" : "boxShadow"}: [
        BoxShadow(
           $offSet,${color}blurRadius: $blurR,
        ),
      ], 
      """;
    }
    return "";
  }
}

File: flutter_code_generate/src/functions/widget/models/submodels/shadow.js
const { fixDouble, hexColorToFlutterColor, sz } = require("../../util");
var withDivision = require("../../main");

class Shadow {
  constructor(json, withColor) {
    this.visible = json['visible'] && withColor;
    if (this.visible) {
      this.x = fixDouble(json['x']);
      this.y = fixDouble(json['y']);
      this.opacity = fixDouble(json['opacity']);
      this.color = hexColorToFlutterColor(json['color'], this.opacity);
      this.blurRadius = fixDouble(json['blur']);
    }
  }

  print(isText) {
    isText = isText == null ? false : isText;
    if (this.visible) {
      let offSet = `offset: Offset(${sz(this.x)},${sz(this.y)})`;
      let blurR = `${sz(this.blurRadius)}`;
      if (withDivision.withDivision)
        return `..boxShadow(${this.color}, blur: ${blurR}, ${offSet})`;
      return `
    ${ isText ? "shadows" : "boxShadow"}: [
      BoxShadow(
        ${offSet}, ${this.color}blurRadius: ${blurR},
      ),
    ],
    `;
    }
    return "";
  }
}

module.exports = { Shadow };
